.chloggen/spanmetricsconnector-disabled.yaml

```diff
@@ -0,0 +1,21 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```

Comment:
Can we make this in two steps? First we add the option with the default value that corresponds to the same behavior. Then, if more people run into this issue, we change the default behavior

Reply:
> How should we let user know that this behavior will change?

We can add this as `subtext` on the release note. I will defer to @albertteoh on whether we should say we are changing this for sure or wait until we have more users complaining about this :)

Reply:
The reasons why I prefer to have "enabled" in sooner rather than later are:
- I prefer the default behaviour of spanmetrics connector to be the bare minimum setup for the basic span to metrics aggregation functionality, with any extraneous functionality such as exemplars being an opt-in. It's a lesson learnt for me too when new features are introduced to consider making them opt-in.
  - I strongly prefer not to have a `disabled` config, because there'd be an inconsistent experience when we add new opt-in functionality that is toggled via an `enabled` flag. Moreover, `enabled` is arguably more idiomatic configuration expression.
- Having `enabled = true` by default, while being consistent with existing behaviour, would be a quite surprising. IMO, it's the same as introducing a `disabled` config.
- As I mentioned in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23872#issuecomment-1625056068, this component is still in [alpha](https://github.com/open-telemetry/opentelemetry-collector#alpha) with the following in its definition: "The configuration options might change often without backwards compatibility guarantees.". As such, I'm expecting we have the liberty of making breaking changes now (within reason), rather than when the component is declared "stable", where I would definitely not advise making this breaking change.
- While it's a breaking change, it's not too onerous for users to modify their config to go back to the previous behaviour, which is well-documented in the change log in this PR by @povilasv.

Reply:
Thanks for explaining, I am fine with marking this with `enabled` false on a single PR since this component is in alpha and the spanmetrics processor was marked as in development before this :+1: 

I do want to point out though that `enabled=true` is the default for some of our core configuration: https://github.com/open-telemetry/opentelemetry-collector/blob/e1d0f265591d484e95eae942c0164c9b16046c8e/exporter/exporterhelper/queued_retry.go#L50 and as such I think is an acceptable pattern